### PR TITLE
Wazowski 23-19 VIA Support

### PR DESCRIPTION
**Pull request by keebzdotnet**
Support for VIA on the Wazowski 23-19 board

## Description

The community asked for VIA support and I've made and tested it. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebzdotnet/wazowski/config.h [hunk 1]
```diff
@@ -20,7 +20,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID 0xFEED
+#define VENDOR_ID 0xKZDN
```
Comment (fauxpark) [inline keyboards/keebzdotnet/wazowski/config.h:23]: This must be a hexadecimal number (0-9, A-F). The defacto standard is to use two ASCII characters instead:
```suggestion
#define VENDOR_ID 0x4B5A // "KZ"
```

Comment (keebzdotnet) [reply keyboards/keebzdotnet/wazowski/config.h:23]: oops thanks for that catch!

File: keyboards/keebzdotnet/wazowski/info.json [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
     "width": 9,
     "height": 5,
     "layouts": {
         "LAYOUT": {
             "layout": [
-                {"label":"k00", "x":0, "y":3},
-                {"label":"k01", "x":1, "y":0},
-                {"label":"k02", "x":1, "y":4},
-                {"label":"k03", "x":2, "y":3},
+                {"label":"k01", "x":1, "y":1},
```
Comment (fauxpark) [inline keyboards/keebzdotnet/wazowski/info.json:10]: The ordering here is kinda strange (testing in Configurator Ctrl+Shift+I with fast input enabled). Is that intentional?

File: keyboards/keebzdotnet/wazowski/rules.mk [hunk 3]
```diff
@@ -14,11 +14,12 @@ BOOTLOADER = atmel-dfu
 # Build Options
 #   change yes to no to disable
 #
-BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+VIA_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/keebzdotnet/wazowski/rules.mk:17]:
```suggestion
```
This should not be enabled at the keyboard level.

File: keyboards/keebzdotnet/wazowski/info.json (hunk 2 above)
Comment (keebzdotnet) [reply keyboards/keebzdotnet/wazowski/info.json:10]: yeah its intentional and looks really weird but this is working and live already for people to use.

File: keyboards/keebzdotnet/wazowski/rules.mk (hunk 3 above)
Comment (keebzdotnet) [reply keyboards/keebzdotnet/wazowski/rules.mk:17]: where should this be enabled at?

Comment (fauxpark) [reply keyboards/keebzdotnet/wazowski/rules.mk:17]: In the `via` keymap's rules.mk - but you're already done that. It just shouldn't be enabled for *all* keymaps.

File: keyboards/keebzdotnet/wazowski/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+enum layer_names {
+    _BASE
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        KC_A, KC_B, KC_C,KC_D, KC_E,
+        KC_F, KC_G, KC_H, KC_I, KC_J,
+        KC_K, KC_L, KC_NO, KC_M, KC_N
```
Comment (zvecr) [inline keyboards/keebzdotnet/wazowski/keymaps/via/keymap.c:28]:
```suggestion
        KC_K, KC_L, KC_NO, KC_M
```

Comment (thompson-ele) [reply keyboards/keebzdotnet/wazowski/keymaps/via/keymap.c]: @zvecr hello! I was able to compile and test this code and it had been working. I think that maybe I was using an outdated version of QMK or something. I've pushed a commit to fix it. I've tested it on my built Wazowski 23-19 and its working. This is my first time writing code for QMK and VIA lol

**Comment by thompson-ele** [comment]
@fauxpark @zvecr I believe we've pushed commits addressing your requested changes. I'm new to the QMK PR process so I'm not sure if we need to add a label to have it reviewed? Or what the usual PR process looks like? :) Thanks for your reviews so far!

File: keyboards/keebzdotnet/wazowski/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        KC_A, KC_B, KC_C, KC_D, KC_E,
+        KC_F, KC_G, KC_H, KC_I, KC_J,
+        KC_K, KC_L, KC_M, KC_N
+    ),
+};
```
Comment (drashna) [inline keyboards/keebzdotnet/wazowski/keymaps/via/keymap.c:30]: If this supports more than one layer, you need to populate each layer, or the keymap will be populated with random data, when the VIA keymap is flashed on a new controller. 

```suggestion

const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
    /* Base */
    [0] = LAYOUT(
        KC_A,    KC_B,    KC_C,    KC_D,    KC_E,
        KC_F,    KC_G,    KC_H,    KC_I,    KC_J,
        KC_K,    KC_L,    KC_M,    KC_N
    ),
    [1] = LAYOUT(
        _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______,
        _______, _______, _______, _______
    ),
    [2] = LAYOUT(
        _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______,
        _______, _______, _______, _______
    ),
    [3] = LAYOUT(
        _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______,
        _______, _______, _______, _______
    ),
};
```

Comment (thompson-ele) [reply keyboards/keebzdotnet/wazowski/keymaps/via/keymap.c]: Thank you @drashna ! Just pushed an update with your suggested changes

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Pull request merged**